### PR TITLE
Remove non-relevant tests and remove the documentation related to them

### PR DESCRIPTION
**Pull request by MatonAnthony**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,4 +8,3 @@ before_install:
 env:
   global:
   - secure: "EIwektOzTBqWLjvRLwMnAwCSExuwxtql+2OLGB+6VU7lOd1sOgIt/UZBru3hdNUTOOhY8TxVdZ3kU611m5We0wS161tNs6AcOw3YlPzHq5SX8cCb3EISVBnDLzKoYf8SGwJMxvnMTnTKmujwEHtmOSHSagN3qUQYo7UQ3GWsP/hnqL9dzbPZuiT/FwuJN1aCrqUxsv+f2a3jBaGHSDSdgT21bu0p6XGzvl4CA8BsN69IJY9OSe/Dn1vy3HTxQkGc8CZEzpH6MpdOLDdL6CwB1JLo8svdX78Zqwv54oUtEvjqgweKgZWYXpQksC310k76aZ4gZsc7+W3bXLKp/Ue8mX8zguGWaoqP8eBw/TYUVi0EY/ubLSPTCFVzT1icj9ZpqBTmO++xU99Di8IAaoJvUP6YN8ppXpsukB+FO+KZJKEhQRM2V6j7BQahz6fXKNK/HooY+EHXoWw24z014chYraaXu88lqepHCg7eiXcIqnazWp0gp2NcEqMNsSf2e0GPIPf1rVjRHOb30dyjU4m/xNI21QQiElYU2TlEqKdJT38fIOErmTDHN6tkNagxngwYzFOOdn+nONESdwXMgw6coZfvLKuj5vnzWZyCKSvkg87bW82CEblJ0Ju2QR+qbzDuVmD3GJKXKOB16Dleyvrol/ofG5+0Vb67CAVphN7OI4s="
```
Comment (Elchi3) [inline .travis.yml]: Do we still need this key? Is it for coveralls which we don't use anymore?

**Comment by Elchi3** [comment]
Developing.md also needs some updating. (remove mentions of karma in the "testing" section, for example)

File: .travis.yml (hunk 1 above)
Comment (MatonAnthony) [inline .travis.yml]: I think it's only used by coveralls, but I don't really know if we used it for anything else.

I can remove it if you deem it appropriate

Comment (Elchi3) [inline .travis.yml]: I think it is not needed anymore.

Comment (MatonAnthony) [inline .travis.yml]: Then i'll remove it.

**Pull request merged**